### PR TITLE
ZEP-18: Add pipeline

### PR DESCRIPTION
**Pull request by cc-nadzir**
This repo comes with a `cloudbuild.yaml`. But it takes time to run due to the docker build of all 10 images.

This PR adds a `cloudbuild-kubectl.yaml` file to deploy the applications with pre-built container images

### CODE REVIEW

**Review by cpwc — APPROVED** [review]

**Review by dshelley66 — COMMENTED** [review]
@cc-nadzir This will need to be done in the context of the cldcvr.yaml format that is being defined by @petmal  - probably you guys should sync up on this. 

One other Q - I thought the microservices-demo used skaffold to build/deploy?

**Comment by cc-nadzir** [comment]
> @cc-nadzir This will need to be done in the context of the cldcvr.yaml format that is being defined by @petmal - probably you guys should sync up on this.
> 
> One other Q - I thought the microservices-demo used skaffold to build/deploy?

sure. will sync up with him.

oh there's several options to deploy the app.
the `kubectl` seems to be the fastest as there is no need to rebuild the docker images

```
Option 3: Using Pre-Built Container Images
💡 Recommended if you want to deploy the app faster in fewer steps to an existing cluster.
```

**Comment by dshelley66** [comment]
> oh there's several options to deploy the app.
> the `kubectl` seems to be the fastest as there is no need to rebuild the docker images
> 
> ```
> Option 3: Using Pre-Built Container Images
> 💡 Recommended if you want to deploy the app faster in fewer steps to an existing cluster.
> ```
Right - I guess we need to find out if the requirement is deployment time or if we are trying to show that the containers are being built from scratch and then deployed - need to clarify this with @vizkid

File: cloudbuild-kubectl.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+substitutions:
```
Comment (petmal) [inline cloudbuild-kubectl.yaml]: In order for this to be usable in Vanguard pipelines the snippet (i.e. pipeline module) must be enriched with some metadata (used by the Vanguard itself and the UI). See the pipeline module definition and example: https://cldcvr.atlassian.net/wiki/spaces/VAN/pages/117211232/Vanguard+pipeline+modules

The code you have here will go to the template section of the module. The constant variables will be replaced by template variables (e.g. `{{ gke_cluster_name }}`) defined in the module's input section.

Once you have the module definition written you can use this tool to render it and test it: https://github.com/cldcvr/vanguard-api/tree/master/pipeline-playground
Just checkout vanguard-api project and follow the readme.md

If you want we can schedule a call tmr.

**Comment by cc-nadzir** [comment]
moving this to vanguard repo

**Pull request closed without merging**